### PR TITLE
#3314 - Introducing filters for Groups and Users listing in identity server with SCIM2

### PR DESCRIPTION
**Pull request by shyabithd**
will be tokenized from '/'

### CODE REVIEW

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sameeragunarathne) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java]: should be the latest license year. 2018

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        super(r);
+    }
+
+    @Override
+    public void parseNumbers() {
+
+    }
+}
+
```
Comment (Megala21) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java]: Remove this extra line

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public CustomStreamTokenizer(Reader r) {
+
+        super(r);
+    }
+
+    @Override
+    public void parseNumbers() {
+
+    }
+}
```
Comment (sameeragunarathne) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java]: add a new line at the end of the file

File: modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+package org.wso2.charon3.core.utils.codeutils;
+
+import java.io.Reader;
+import java.io.StreamTokenizer;
+
+/**
+ * This class is basically an extension of Stream tokenizer class which is used to tokenize strings
+ * filter (eg : id eq 12334fd-34vfd) expression needs to be parsed 12334fd-34vfd as value. otherwise numbers will be
+ * taken separately
```
Comment (sameeragunarathne) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/utils/codeutils/CustomStreamTokenizer.java]: Java doc comments should end with the period.

**Comment by shyabithd** [comment]
@omindu I created an issue in product IS (Introducing filters for Groups and Users listing in identity server with SCIM2 #3314) and linked this ticket to created one.

**Pull request merged**